susemanager-utils/susemanager-sls/src/modules/appstreams.py

```diff
@@ -92,6 +92,26 @@ def _get_module_info(module_names):
         return []
 
     module_info_output = result.stdout.splitlines()
+    active_modules = []
+    while True:
+        start = next(
+            (i for i, e in enumerate(module_info_output)
+            if "[a]" in e), -1
+        )
+        if start != -1 and module_info_output[start].startswith("Stream"):
+            end = next(
+                (i for i, e in enumerate(module_info_output[start:-1])
+                if e.startswith("Name")), -1
+            )
+            if end != -1:
+                active_modules+=module_info_output[start-1:end+start]
+                module_info_output = module_info_output[end+start:-1]
+            else:
+                active_modules+=module_info_output[start-1:end]
+                module_info_output=active_modules
+                break
+        else:
+            break
```

Comment:
I would probably rather prefer to iterate `module_info_output` outside of a `while True` loop, for example, iterating `module_info_output` line-by-line (if that makes sense), and then filtering and collecting only the content of active modules we want.

Reply:
Sorry for the bad explanation! I was thinking on something a bit more easy to follow, maybe using "regex", like:

```python
    module_info_output = []
    for module in re.findall(r"(Name\s+:.*?)(?=\n\s*\n|$)", result.stdout, re.DOTALL):
        if re.search(r"Stream\s+:.*\[a\]", module):
            module_info_output += module.splitlines()
```